src/main/java/com/cloudbees/hudson/plugins/folder/icons/BuildStatusFolderIcon.java

```diff
@@ -0,0 +1,84 @@
+package com.cloudbees.hudson.plugins.folder.icons;
+
+import com.cloudbees.hudson.plugins.folder.Folder;
+import com.cloudbees.hudson.plugins.folder.FolderIcon;
+import com.cloudbees.hudson.plugins.folder.FolderIconDescriptor;
+import hudson.Extension;
+import hudson.model.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.Stapler;
+
+public class BuildStatusFolderIcon extends FolderIcon {
+
+    Folder folder;
+
+    @DataBoundConstructor
+    public BuildStatusFolderIcon() {
+    }
+
+    public Integer ordinal() {
+      return this.getCombinedBallColor().ordinal();
+    }
+
+    private BallColor getCombinedBallColor() {
+        Boolean taskIsBuilding = false;
+        Boolean oneTaskIsIncomplete = false;
+        Result combinedResult = null;
+        for(Job job : this.folder.getAllJobs()) {
+            Run build = job.getLastBuild();
+            if(build == null) {
+                oneTaskIsIncomplete = true;
+                continue;
```

Comment:
So it seems like perhaps either this should be a `break` or line 40 should be a `continue`.

The way this will work currently is it will set `taskIsBuilding` *only* if the first job with a build is building. I would expect `taskIsBuilding` to be set if *any* job is building.

Seems like there is a logic bug here.